File: client/lib/annoucements/announcement.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:pet_share/annoucements/pet.dart';
import 'package:pet_share/shelter.dart';

@JsonSerializable()
class Announcement {
  Announcement({
    required this.pet,
    required this.title,
    required this.description,
    required this.shelter,
  });

  Pet pet;
  String title;
  String description;
  Shelter shelter;
  AnnouncementStatus status = AnnouncementStatus.open;
}

enum AnnouncementStatus {
  open,
  closed,
  removed,
  inVerification,
}

File: apis/DatabaseContextLibrary/Models/Announcement.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace DatabaseContextLibrary.models
{
    public class Announcement
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public Guid Id { get; set; }
        public string? Title { get; set; }
        public string? Description { get; set; }
        public DateTime CreationDate { get; set; }
        public DateTime? ClosingDate { get; set; }
        public DateTime LastUpdateDate { get; set; }
        public Status Status { get; set; }
        public Guid PetId { get; set; }
        public Pet Pet { get; set; }
    }

    public enum Status
    {
        Open,
        Closed,
        DuringVerification,
        Deleted
    }
}
